### PR TITLE
Update the advanced settings editor tab

### LINKED ISSUE #3210: [FEATURE] Allow deleting settings from the Advanced tab in options
Original commit: https://github.com/qgis/QGIS/commit/5349f24cae83219d783e80e94f198a7cb421bee8 by nyalldawson

This commit adds a new right click menu to the settings shown
in the "Advanced" tab in the settings dialog, which allows users
to remove that setting (or group of settings)

### LINKED ISSUE #3541: Make default geometry checks configurable via settings [needs-docs]
Original commit: https://github.com/qgis/QGIS/commit/1645799788b9ed1b5f40613bc8360d1e365cad2e by m-kuhn

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by DelazJ**
Fix #3541
And mention it's possible to remove settings (fix #3210)

### CODE REVIEW

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 1]
```diff
@@ -899,31 +899,32 @@ if a very slow search filter is present (e.g. one which uses an online service).
...
-means to manage these settings through the :guilabel:`Advanced Settings Editor`.
-After you promise to be careful,
-the widget is populated with a tree of all QGIS settings, which you can directly edit.
+   Advanced Settings tab in QGIS
+
+All the settings related to QGIS (UI, tools, data providers, Processing
+configurations, default values and paths, plugins options, expressions,
+geometry checks...) are saved in a :file:`QGIS/QGIS3.ini` file under the active
+:ref:`user profile <user_profiles>` directory.
+Pasting this file among QGIS installations also share the configurations.
```
Comment (havatv) [inline source/docs/user_manual/introduction/qgis_configuration.rst]:
```suggestion
Configurations can be shared by copying this file to other installations.
```

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -899,31 +899,32 @@ if a very slow search filter is present (e.g. one which uses an online service).
...
+
+From within QGIS, the :guilabel:`Advanced` tab offers a way to manage these
+settings through the :guilabel:`Advanced Settings Editor`.
+After you promise to be careful, the widget is populated with a tree of all
+the existing settings, whose value you can directly edit.
```
Comment (havatv) [inline source/docs/user_manual/introduction/qgis_configuration.rst]:
```suggestion
the settings, and these can all be modified.
```

File: source/docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -899,31 +899,32 @@ if a very slow search filter is present (e.g. one which uses an online service).
...
+Right-click over a setting or a group and you can delete it.
```
Comment (havatv) [inline source/docs/user_manual/introduction/qgis_configuration.rst]:
```suggestion
Right-click over a setting or a group and you can delete it
(to add a setting or group, you have to edit the QGIS3.ini file).
```

File: source/docs/user_manual/introduction/qgis_configuration.rst (hunk 2 above)
Comment (DelazJ) [reply source/docs/user_manual/introduction/qgis_configuration.rst]: hmmm... Issue here is that the "tree" has four columns and only the value column can be modified; you can not change setting name, type or fill a description, so claiming that *these can all be modified*, *these* not indexing anything particular, is a bit too much imho.

Comment (havatv) [reply source/docs/user_manual/introduction/qgis_configuration.rst]: I gave it another go.  Not sure if it is better, though...

**Pull request merged**